backend/s3/s3.go

```diff
@@ -2177,6 +2233,39 @@ func (f *Fs) makeBucket(ctx context.Context, bucket string) error {
 //
 // Returns an error if it isn't empty
 func (f *Fs) Rmdir(ctx context.Context, dir string) error {
+	// Add trailing slash to folder path
+	dirFilePath := fmt.Sprintf("%s/", dir)
+	if len(f.rootDirectory) != 0 {
+		dirFilePath = fmt.Sprintf("%s/%s", f.rootDirectory, dirFilePath)
+	}
+	// Query directory marker object before attempting to remove it.
+	httpReq, _ := f.c.GetObjectRequest(&s3.GetObjectInput{
+		Bucket: &f.rootBucket,
+		Key:    &dirFilePath,
+	})
+	err := f.pacer.Call(func() (bool, error) {
+		var err error
+		httpReq.HTTPRequest = httpReq.HTTPRequest.WithContext(ctx)
+		err = httpReq.Send()
+		return f.shouldRetry(ctx, err)
+	})
+	if err == nil {
+		// Using API call due to trailing slash being removed before Remove() can get full path.
```

Comment:
It would be better to call Object.Remove here rather than duplicating the code if possible.

Reply:
Done, indeed looks better. This one missed `DirectoryMarkers` conditional that caused them always to be removed.